test/extended/router/config_manager.go

```diff
@@ -516,15 +516,15 @@ http {
 
 			g.By("waiting for the healthz endpoint to respond")
 			healthzURI := fmt.Sprintf("http://%s/healthz", net.JoinHostPort(routerIP, "1936"))
-			err = waitForRouterOKResponseExec(ns, execPod.Name, healthzURI, routerIP, timeoutSeconds)
+			err = waitForRouterOKResponseExec(ns, execPod.Name, healthzURI, exutil.IPUrl(routerIP), timeoutSeconds)
```

Comment:
Can the underlying functions be adjusted instead? My preference is to not make the caller handle this.  You're far more likely to fix the problem for good if you do it in `waitForRouterOKResponseExec` and whatever functions are building the URL's themselves.

Reply:
I think the utility exutil.IPUrl is fine, but can you make it so the body of these functions wraps the IPv6 addresses instead? I've been chasing IPv6 bugs for a few years and I guarantee you more people are going to try to call these wait functions and will pass unbracketed IP's into it.

i.e.

```
func waitForRouterOKResponseExec(ns, execPodName, url, host string, timeoutSeconds int) error {
        host := exutil.IPUrl(routerIP)
	cmd := fmt.Sprintf(`
		set -e
		STOP=$(($(date '+%%s') + %d))
		while [ $(date '+%%s') -lt $STOP ]; do
			rc=0
			code=$( curl -k -s -m 5 -o /dev/null -w '%%{http_code}\n' --header 'Host: %s' %q ) || rc=$?
			if [[ "${rc:-0}" -eq 0 ]]; then
				echo $code
				if [[ $code -eq 200 ]]; then
					exit 0
				fi
				if [[ $code -ne 503 ]]; then
					exit 1
				fi
			else
				echo "error ${rc}" 1>&2
			fi
			sleep 1
		done
		`, timeoutSeconds, host, url)
	output, err := e2e.RunHostCmd(ns, execPodName, cmd)
	if err != nil {
		return fmt.Errorf("host command failed: %v\n%s", err, output)
	}
	lines := strings.Split(strings.TrimSpace(output), "\n")
	if lines[len(lines)-1] != "200" {
		return fmt.Errorf("last response from server was not 200:\n%s", output)
	}
	return nil
}
```